service/src/main/kotlin/fi/espoo/evaka/messaging/message/MessageController.kt

```diff
@@ -225,8 +230,16 @@ class MessageController(
             ?: throw Forbidden("Message account not found for user")
     }
 
-    @GetMapping
-    fun getThreadsMock(): Paged<MessageThread> {
-        return mockThreadData()
+    private fun checkAuthorizedRecipients(
+        db: Database.Connection,
+        user: AuthenticatedUser,
+        recipientAccountIds: Set<UUID>
+    ) {
+        db.read {
+            it.isUserAuthorizedToSendTo(
+                user,
+                recipientAccountIds
+            )
+        } || throw Forbidden("Not authorized to send to the given recipients")
```

Comment:
This reminded me that frontend error handling needs to be properly implemented. Added a task to board.